### PR TITLE
GEODE-8540: Create new DistributedBlackboard Rule

### PR DESCRIPTION
**Pull request by kirklund**
Package up DUnitBlackboard as a JUnit Rule named DistributedBlackboard.

### CODE REVIEW

File: geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/DistributedBlackboardDistributedTest.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Test
+  public void canPassDataBetweenVMs() {
+    VM vm0 = getVM(0);
+    VM vm1 = getVM(1);
+
+    vm0.invoke("put data in mailbox", () -> blackboard.setMailbox(mailbox(), value()));
+
```
Comment (demery-pivotal) [inline geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/DistributedBlackboardDistributedTest.java:51]: You have no idea how much I appreciate you naming these messages. Thank you!

File: geode-dunit/src/main/java/org/apache/geode/test/dunit/rules/DistributedBlackboard.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.geode.test.dunit.Blackboard;
+import org.apache.geode.test.dunit.DUnitBlackboard;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.internal.InternalBlackboard;
+import org.apache.geode.test.dunit.internal.InternalBlackboardImpl;
+
+/**
+ * BlackboardRule provides mailboxes and synchronization gateways for distributed tests.
```
Comment (demery-pivotal) [inline geode-dunit/src/main/java/org/apache/geode/test/dunit/rules/DistributedBlackboard.java:31]: DistributedBlackboard

**Review by demery-pivotal — APPROVED** [review]

File: geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/DistributedBlackboardDistributedTest.java (hunk 1 above)
Comment (kirklund) [reply geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/DistributedBlackboardDistributedTest.java:51]: I copied that from Bruce's original test ;)

File: geode-dunit/src/main/java/org/apache/geode/test/dunit/rules/DistributedBlackboard.java (hunk 2 above)
Comment (kirklund) [reply geode-dunit/src/main/java/org/apache/geode/test/dunit/rules/DistributedBlackboard.java:31]: Thanks for catching that!

**Pull request merged**